### PR TITLE
Workspace - group of projects

### PR DESCRIPTION
**Pull request by 0xc0170**
generate_dic(self, tool, copy) in Project() is replaced by small functions which has one clear task.

The biggest change here is , I removed -w (workspaces) as it really does not make sense. It's just a group of projects. I implemented it for IAR at the moment.

One project there contains eww file (workspace file) which poitns to a project in the same directory. That's how it was. Now groups come into play. define a workspace like this

```
projects:
    group_test:
        k20d50m_blinky: &k20d50m_blinky
            - *module_mbed
            - tools/records/mbed/k20_cmsis.yaml
            - tools/records/mbed/k20_target.yaml
            - tools/records/projects/k20_blinky.yaml
        lpc1768_blinky: &lpc1768_blinky
            - *module_lpc1768
            - *module_mbed
            - *proj_settings
            - tools/records/projects/lpc1768_blinky.yaml
```

The generated structure will be (by default):
generated_projects\group_test\group_test.eww (see one workspace file)
generated_projects\group_test\k20d50m_blinky (project files inside)
generated_projects\group_test\lpc1768_blinky (project files inside)

This needs much more testing, as individual projects can be generated anywhere, so workspace needs to have their paths. I'll be testing it further for IAR at the moment as .eww support was already  there. 

This is not breaking change, just improvements. One question is which I asked already in the issue #192, how to set a path for workspace file. I am now working with the default one. I'll add some comments to that issue to discuss it further

### CODE REVIEW

**Comment by 0xc0170** [comment]
We will need to write tests for workspaces before merging this

**Comment by 0xc0170** [comment]
IAR and uvision workspaces now supported. One main task is , how to define export_dir for workspace file?

File: project_generator/tools/iar.py [hunk 1]
```diff
@@ -135,9 +135,17 @@ def _ewp_set_toolchain(self, ewp_dic, toolchain):
     def _ewp_set_name(self, ewp_dic, name):
         ewp_dic['project']['configuration']['name'] = name
 
-    def _eww_set_path(self, eww_dic, name):
+    def _eww_set_path_single_project(self, eww_dic, name):
```
Comment (sarahmarshy) [inline project_generator/tools/iar.py]: Should we have tool specific functions in the generic project class? Could this maybe be done in tool class?

File: project_generator/tools/iar.py [hunk 2]
```diff
@@ -135,9 +135,17 @@ def _ewp_set_toolchain(self, ewp_dic, toolchain):
...
         eww_dic['workspace']['project']['path'] = join('$WS_DIR$', name + '.ewp')
 
+    def _eww_set_path_multiple_project(self, eww_dic, projects):
+        eww_dic['workspace']['project'] = []
+        for project in self.workspace['projects']:
+            # This has an assumption all projects inside workspace
```
Comment (0xc0170) [inline project_generator/tools/iar.py]: @sarahmarshy I received an email with your comments to this one. 

Yes this should be handlened in the generic layer. I did something similar for uvision, so we see the patern here and just should fix it in the ProjectWorkspace() class.

**Comment by 0xc0170** [comment]
Wanted - workspace tests !

**Pull request merged**